components/org.wso2.carbon.identity.application.authenticator.basicauth/src/main/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticator.java

```diff
@@ -133,9 +129,20 @@ protected AuthenticatorFlowStatus executeAutoLoginFlow(HttpServletRequest reques
                                                          AuthenticationContext context, Cookie autoLoginCookie)
             throws AuthenticationFailedException {
 
-        JSONObject cookieValueJSON = transformCookieValueToJSON(autoLoginCookie);
-        String usernameInCookie = (String) cookieValueJSON.get(USERNAME);
-        String usernameInHttpRequest = request.getParameter(USERNAME);
+        String decodedValue = new String(Base64.getDecoder().decode(autoLoginCookie.getValue()));
+        JSONObject cookieValueJSON = transformToJSON(decodedValue);
+        String usernameInCookie = (String) cookieValueJSON.get(AutoLoginConstant.USERNAME);
+        String signature = (String) cookieValueJSON.get(AutoLoginConstant.SIGNATURE);
+        String content = usernameInCookie;
+        String alias = null;
+        if (StringUtils.isEmpty(usernameInCookie)) {
+            content = (String) cookieValueJSON.get(AutoLoginConstant.CONTENT);
+            JSONObject contentJSON = transformToJSON(content);
+            usernameInCookie = (String) contentJSON.get(AutoLoginConstant.USERNAME);
+            alias = getSelfRegistrationAutoLoginAlias(context);
```

Comment:
What happens to the alias if usernameInCookie is not empty? Dont we need alias in that case?